posthog/queries/app_metrics/app_metrics.py

```diff
@@ -0,0 +1,77 @@
+from datetime import timedelta
+
+from django.utils.timezone import now
+
+from posthog.client import sync_execute
+from posthog.models.app_metrics.sql import QUERY_APP_METRICS_TIME_SERIES
+from posthog.models.filters.mixins.base import IntervalType
+from posthog.models.team.team import Team
+from posthog.queries.app_metrics.serializers import AppMetricsRequestSerializer
+from posthog.queries.util import format_ch_timestamp
+from posthog.utils import relative_date_parse
+
+
+class AppMetricsQuery:
+    def __init__(self, team: Team, plugin_config_id: int, filter: AppMetricsRequestSerializer):
+        self.team = team
+        self.plugin_config_id = plugin_config_id
+        self.filter = filter
+
+    def run(self):
+        query, params = self.metrics_query()
+        dates, successes, successes_on_retry, failures = sync_execute(query, params)[0]
+        return {
+            "dates": [
+                timestamp.strftime("%Y-%m-%d{}".format(" %H:%M:%S" if self.interval == "hour" else ""))
+                for timestamp in dates
+            ],
+            "successes": successes,
+            "successes_on_retry": successes_on_retry,
+            "failures": failures,
+            "totals": {
+                "successes": sum(successes),
+                "successes_on_retry": sum(successes_on_retry),
+                "failures": sum(failures),
+            },
+        }
+
+    def metrics_query(self):
+        job_id = self.filter.validated_data.get("job_id")
+        query = QUERY_APP_METRICS_TIME_SERIES.format(
+            job_id_clause="AND job_id = %(job_id)s" if job_id is not None else "",
+            interval_function=self.interval_function,
+        )
+
+        return query, {
+            "team_id": self.team.pk,
+            "plugin_config_id": self.plugin_config_id,
+            "category": self.filter.validated_data.get("category"),
+            "job_id": job_id,
+            "date_from": format_ch_timestamp(self.date_from, self.team.timezone),
+            "date_to": format_ch_timestamp(self.date_to, self.team.timezone),
+            "timezone": self.team.timezone,
+            "interval": self.interval,
+        }
+
+    @property
+    def date_from(self):
+        return relative_date_parse(self.filter.validated_data.get("date_from"))
+
+    @property
+    def date_to(self):
+        date_to_string = self.filter.validated_data.get("date_to")
+        return relative_date_parse(date_to_string) if date_to_string is not None else now()
+
+    @property
+    def interval(self) -> IntervalType:
+        if self.date_to - self.date_from < timedelta(days=2):
```

Comment:
I'd let the api user decide this for their query

Reply:
This is used potentially in two places:
1. Showing metrics on a single historical export
2. (Potentially) Allowing users to choose "past 24 hours" in a dropdown (we default to 30d).

In neither case does it make sense to make this an API param, especially for (1)